pkg/network/protocols/events/consumer.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package events
+
+import (
+	"fmt"
+	"sync"
+	"time"
+	"unsafe"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	manager "github.com/DataDog/ebpf-manager"
+	"go.uber.org/atomic"
+)
+
+const (
+	batchMapSuffix  = "_batches"
+	eventsMapSuffix = "_batch_events"
+)
+
+// Consumer provides a standardized abstraction for consuming (batched) events from eBPF
+type Consumer struct {
+	proto       string
+	syncRequest chan (chan struct{})
+	offsets     *offsetManager
+	handler     *ddebpf.PerfHandler
+	batchReader *batchReader
+	callback    func([]byte)
+
+	// termination
+	mux         sync.Mutex
+	eventLoopWG sync.WaitGroup
+	stopped     bool
+
+	// telemetry
+	then        time.Time
+	eventsCount *telemetry.Metric
+	missesCount *telemetry.Metric
+	batchSize   *atomic.Int64
+}
+
+// NewConsumer instantiates a new event Consumer
+// `callback` is executed once for every "event" generated on eBPF and must:
+// 1) copy the data it wishes to hold since the underlying byte array is reclaimed;
+// 2) be thread-safe, as the callback may be executed concurrently from multiple go-routines;
+func NewConsumer(proto string, ebpf *manager.Manager, callback func([]byte)) (*Consumer, error) {
+	batchMapName := proto + batchMapSuffix
+	batchMap, found, _ := ebpf.GetMap(batchMapName)
+	if !found {
+		return nil, fmt.Errorf("unable to find map %s", batchMapName)
+	}
+
+	eventsMapName := proto + eventsMapSuffix
+	eventsMap, found, _ := ebpf.GetMap(eventsMapName)
+	if !found {
+		return nil, fmt.Errorf("unable to find map %s", eventsMapName)
+	}
+
+	numCPUs := int(eventsMap.MaxEntries())
+	offsets := newOffsetManager(numCPUs)
+	batchReader, err := newBatchReader(offsets, batchMap, numCPUs)
+	if err != nil {
+		return nil, err
+	}
+
+	handlerMux.Lock()
+	handler, ok := handlerByProtocol[proto]
+	delete(handlerByProtocol, proto)
```

Comment:
why?

Reply:
Let me know if the comments and changes to `configuration.go` don't address your question